### PR TITLE
account for empty nse in rawio.neuralynxrawio

### PR DESCRIPTION
**Pull request by sin-mike**
Guys, I've made an update for a workaround with the empty nse files
There still will be info in self._spike_memmap, but it will be just empty numpy arrays

### CODE REVIEW

File: neo/rawio/neuralynxrawio.py [hunk 1]
```diff
@@ -144,7 +145,17 @@ def _parse_header(self):
...
                     dtype = get_nse_or_ntt_dtype(info, ext)
-                    data = np.memmap(filename, dtype=dtype, mode='r', offset=HEADER_SIZE)
+
+                    if (os.path.getsize(filename)<=HEADER_SIZE):
+                        self._empty_nse_ntt.append(filename)
+                        data = dict(unit_id    = np.array([]), 
+                                    channel_id = np.array([]), 
+                                    timestamp  = np.array([]), 
+                                    features   = np.array([[]]), 
+                                    samples    = np.array([[]]))
```
Comment (samuelgarcia) [inline neo/rawio/neuralynxrawio.py]: Here I would prefer :
```python 
data = np.zeros((0,), dtype=dtype)
```

File: neo/rawio/neuralynxrawio.py [hunk 2]
```diff
@@ -89,7 +90,7 @@ def _parse_header(self):
             if ext not in self.extensions:
                 continue
 
-            if (os.path.getsize(filename)<=HEADER_SIZE) and (ext == 'ncs'):
+            if (os.path.getsize(filename)<=HEADER_SIZE) and (ext in ['ncs']):
```
Comment (samuelgarcia) [inline neo/rawio/neuralynxrawio.py]: Do you mean
```python
(ext in ['ncs', 'ntt']):
```

**Comment by samuelgarcia** [comment]
Thank you for the patch.
I have made 2 comments.

Unfortunatly there are a lot of there weird pep8 **invalid escape sequence '\S'** but it was already in the original code.

File: neo/rawio/neuralynxrawio.py (hunk 2 above)
Comment (sin-mike) [reply neo/rawio/neuralynxrawio.py]: At first, there were 
`ext in ['ncs', 'nse', 'ntt']`
but then we had a talk in #553, and then I just put 'ncs'.  
But this version of mine lived in my local verison. Now I just wanted to sync with the master. 

So afaik, there were 2 reasons:
1. Empty NCS are not important
2. Empty spikes (NSE, NTT) are somehow important. However, we didn't ever used this info.

Therefore, I just skipped empty ncs and put empty info to nse and ntt

Comment (samuelgarcia) [reply neo/rawio/neuralynxrawio.py]: Oups.
I reply to quickly.
I forgot the last patch and mix nse and nse.
I am sorry, this is OK.

File: neo/rawio/neuralynxrawio.py [hunk 3]
```diff
@@ -148,11 +148,11 @@ def _parse_header(self):
...
-                        data = dict(unit_id    = np.array([]), 
-                                    channel_id = np.array([]), 
-                                    timestamp  = np.array([]), 
-                                    features   = np.array([[]]), 
-                                    samples    = np.array([[]]))
+                        data = dict(unit_id    = np.zeros((0,), dtype=dtype), 
+                                    channel_id = np.zeros((0,), dtype=dtype), 
+                                    timestamp  = np.zeros((0,), dtype=dtype), 
+                                    features   = np.zeros((0,0), dtype=dtype), 
+                                    samples    = np.zeros((0,0), dtype=dtype))
```
Comment (samuelgarcia) [inline neo/rawio/neuralynxrawio.py]: Oups.
My proposal was:
```data = np.zeros((0,), dtype=dtype)```
would replace all lines 151 - 155
data must be numpy .array 1D  (memmap or not) with a structured dtype not a dict.

Comment (sin-mike) [reply neo/rawio/neuralynxrawio.py]: done)

**Comment by samuelgarcia** [comment]
OK for me.
I think we can merge.

**Pull request merged**